docworks-dts/lib/multiple-files/repoCreator.js

```diff
@@ -0,0 +1,50 @@
+const { createHierarchicalServicesMap } = require('./serviceUtils')
+const { moduleBuilder, $wGlobalNamespaceBuilder } = require('./builders')
+const { $W_NAME } = require('./constants')
+const { getModulesDependenciesMap } = require('./providers/modulesDependencies')
+const {
+	emit,
+	dtsImportDefault
+} = require('./generators')
+
+const is$w = name => name === $W_NAME
+
+const createModulesFilesMap = ({ services, run$wFixer }) => {
+	const servicesMap = createHierarchicalServicesMap(services)
+	return Object.keys(servicesMap).reduce((filesMap, moduleName) => {
+		const rootService = servicesMap[moduleName]
+		return {
+			...filesMap,
+			[moduleName]: is$w(moduleName)
+				? $wGlobalNamespaceBuilder(rootService, { run$wFixer })
+				: moduleBuilder(rootService)
+		}
+	}, {})
+}
+
+const createImportStatements = dependencies => {
+	return Object.keys(dependencies).map(depModuleKey => {
+		const dependency = dependencies[depModuleKey]
+		return emit(dtsImportDefault(dependency.to, dependency.from))
+	})
+}
+
+const enrichModuleContent = (module, extraConetnt) => {
+	if (module) {
+		module.content = [...extraConetnt, module.content].join('\n')
+	}
+}
+
+const repoCreator = ({ services, run$wFixer }) => {
```

Comment:
why did you decide to call this `repoCreator`?
It's not very clear to me.

Reply:
repo is docworks terminology. it represents the part that handles the docs repo (all docworks services that are part of wix-code-docs).. that make more sense?

Reply:
yeah :) sounds good